challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map { |e| e.capitalize }
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select { |e| e == "dog"}
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject { |e| e == "dog"}
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.length
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input[0..1]
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.grep /cat|dog/
```

Comment:
This was tough and took awhile.  At first, I tried to somehow do a find_all? loop, but I couldn't figure out EXACTLY how to make it work.  For example, I tried something like this:

input.find_all do |cditems|
  	cditems = animals.include?("c", "d")
  	end

Despite trying several variations of find_all? , I couldn't get it to work.  I also thought about doing reject where we would reject any items that didn't include "c" or "d" but there's only so much time in the day.

Anyway, I ended up using .select and an include?, which worked.  Notice that I added the two items to the array for a total of 6 animals; the line items at the bottom had to be modified slightly, of course, in order to pass true to all six. It was mandatory that the array started at the first item (0). Trying it at item 1,2,3 etc returned false.  Also, doing ('c', 'd') with a comma didn't work; it had to be done with two dots '..' in order to work (this was annoying).

def fetch_CD_animals(input)
  input.select { |cditems| ('c'..'d').include? cditems[0]}
end

## DO NOT CHANGE CODE BELOW THIS LINE ##

animals = ["cat", "moose", "dog", "bird", "canary", "dingo"]

p capitalize_each_string(animals) == ["Cat", "Moose", "Dog", "Bird", "Canary", "Dingo"]

p fetch_the_dog(animals) == ["dog"] 

p no_dogs_allowed(animals) == ["cat", "moose", "bird", "canary", "dingo"]

p count_the_animals(animals) == 6

p fetch_the_first_two(animals)  == ["cat", "moose"]

p fetch_CD_animals(animals) == ["cat", "dog", "canary", "dingo"]